added AES encryption and also updated the testcases to verify encrypt…

itzVort3x1 opened the pull request:
### `bob2_protocol.py`
- Implemented the encryption and decryption functions in the protocol:
  1. The `encrypt_message` method encrypts the message content using AES-256 in CBC mode, with a randomly generated IV prepended to the ciphertext.
  2. The `decrypt_message` method extracts the IV, decrypts the ciphertext, and removes the padding applied during encryption.

### `test_bob2_protocol.py`
- Added `test_encryption` to verify the encryption functionality.

### NOTE:
I believe the test is failing as I have modified the `setUp` to 
```python 
self.encryption_key = os.urandom(32) 
self.bob2 = Bob2Protocol(encryption_key=self.encryption_key)
```

itzVort3x1 committed:
added AES encryption and also updated the testcases to verify encryption.

itzVort3x1 committed:
commented project path

aryansingh920 commented:
I think you'll have to update the yaml as well if you have done changes with the test/ since we have added an automation workflow to check before merging the PR

itzVort3x1 committed:
Update python-app.yml

itzVort3x1 committed:
updated main.py

itzVort3x1 committed:
updated yml

itzVort3x1 committed:
updated yml to run on any branch

itzVort3x1 commented:
> I think you'll have to update the yaml as well if you have done changes with the test/ since we have added an automation workflow to check before merging the PR

I have updated the yml file to run on any branch.

itzVort3x1 committed:
updated yml to run on pushed branch and main

clairegregg commented:
What is the plan here to ensure only the correct devices can decrypt messages - IE, how will the encryption key be handled? 

Also, could you please add any details of this change to the README and bump the version to represent the change?
